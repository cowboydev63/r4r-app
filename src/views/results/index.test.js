import React from 'react';
import { shallow } from 'enzyme';
import { Results } from './index';

// TODO: window.matchmedia means we need to use jsdom to provide a window for even smoke
describe('Results view', () => {
  beforeAll(() => {  
    Object.defineProperty(window, "matchMedia", {
      value: jest.fn(() => ({ matches: true, addListener: () => {} }) )
    });
  });

  it('renders without crashing', () => {
    shallow(<Results location={{}} setCurrentSearchText={() => {}} validatedNewSearch={()=>{}}/>);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Results location={{search: ''}} setCurrentSearchText={() => {}} validatedNewSearch={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();
  })
})
