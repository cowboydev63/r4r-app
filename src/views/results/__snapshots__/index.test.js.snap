// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results view renders correctly 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Resources for Researchers: Search Results - National Cancer Institute
    </title>
    <meta
      content="Resources for Researchers is a tool to give researchers a better understanding of the various tools available to them."
      property="og:description"
    />
    <meta
      content="https://www.cancer.gov/research/r4r/search"
      property="og:url"
    />
    <meta
      content="Resources for Researchers: Search Results - National Cancer Institute"
      property="twitter:title"
    />
  </HelmetWrapper>
  <CTS_Spinner />
</React.Fragment>
`;
